src/main/c/cext/ruby.c

```diff
@@ -3227,7 +3227,13 @@ VALUE rb_enc_str_new_cstr(const char *ptr, rb_encoding *enc) {
 }
 
 VALUE rb_enc_str_new_static(const char *ptr, long len, rb_encoding *enc) {
-  rb_tr_error("rb_enc_str_new_static not implemented");
+  if (len < 0) {
+    rb_raise(rb_eArgError, "negative string size (or size too big)");
+  }
```

Comment:
Should this check be part of `rb_str_new` instead?

Reply:
MRI does it in `str_new_static`.

Reply:
Exactly, so doing the check as down as we can without touching the pointer yet.
These `_static` variants are just optimizations for MRI, they are useless to us and should just invoke the non-`_static` variant.